workqueue.c

```diff
@@ -113,6 +116,79 @@ void workqueue_enqueue(struct workqueue *wq, struct workqueue_work *work)
 
 	pthread_cond_signal(&sw->cond);
 	pthread_mutex_unlock(&sw->lock);
+
+	return FIO_Q_QUEUED;
+}
+
+/*
+ * Check in the workqueue for overlap and return FIO_Q_BUSY if found.
+ */
+enum fio_q_status workqueue_enqueue_serial_overlap(struct workqueue *wq, struct workqueue_work *work)
+{
+	struct io_u *io_u = container_of(work, struct io_u, work);
+	struct thread_data *td;
+	struct flist_head *entry;
+	struct workqueue *tdwq;
+	int i, j;
+	unsigned long long x1, x2, y1, y2;
+
+	x1 = io_u->offset;
+	x2 = io_u->offset + io_u->buflen;
+
+	pthread_mutex_lock(&serialize_lock);
+
+	for_each_td(td, i) {
```

Comment:
This seems pretty nutty... Why are you iterating every td? Doesn't this for every IO is going to be super slow, depending on what job is being run. You're looping each work item, in each worker, in each td - for every IO.